jupyter_rsession_proxy/handlers.py

```diff
@@ -0,0 +1,165 @@
+# vim: set et sw=4 ts=4:
+import getpass
+import os
+import shutil
+import subprocess
+import tempfile
+from textwrap import dedent
+
+from jupyter_server.utils import url_path_join as ujoin
+from jupyter_server_proxy.handlers import SuperviseAndProxyHandler, AddSlashHandler
+from jupyter_server_proxy.api import IconHandler
+
+def get_rstudio_executable(prog):
+    # Find prog in known locations
+    other_paths = [
+        # When rstudio-server deb is installed
+        os.path.join('/usr/lib/rstudio-server/bin', prog),
+        # When just rstudio deb is installed
+        os.path.join('/usr/lib/rstudio/bin', prog),
+    ]
+    if shutil.which(prog):
+        return prog
+
+    for op in other_paths:
+        if os.path.exists(op):
+            return op
+
+    raise FileNotFoundError(f'Could not find {prog} in PATH')
+
+def get_icon_path():
+    return os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), 'icons', 'rstudio.svg'
+    )
+
+def db_config():
+    '''
+    Create a temporary directory to hold rserver's database, and create
+    the configuration file rserver uses to find the database.
+
+    https://docs.rstudio.com/ide/server-pro/latest/database.html
+    https://github.com/rstudio/rstudio/tree/v1.4.1103/src/cpp/server/db
+    '''
+    # use mkdtemp() so the directory and its contents don't vanish when
+    # we're out of scope
+    db_dir = tempfile.mkdtemp()
+    # create the rserver database config
+    db_conf = dedent("""
+        provider=sqlite
+        directory={directory}
+    """).format(directory=db_dir)
+    f = tempfile.NamedTemporaryFile(mode='w', delete=False, dir=db_dir)
```

Comment:
If db_dir is already a temporary directory, I think it'll be helpful to name the config file deterministically - so we don't have two securely generated tmp file names in path.